app/ChartNotifier.js

```diff
@@ -0,0 +1,22 @@
+
+
+export default class ChartNotifier {
+    messageCallback = null;
+    removeByCategoryCallback=null
+
+    notify(message) {
+        this.messageCallback(message);
+    }
+
+    removeByCategory(category) {
+        this.removeByCategoryCallback(category);
+    }
+
+    setMessageCallback(callback) {
+        this.messageCallback = callback;
+    }
+
+    setRemoveByCategoryCallback(callback) {
```

Comment:
@shayanbinary do you think it should be `onMessage` and `onRemoveByCategory`? I think that's the convention in general for assigning event handlers.